### PR TITLE
feat(core): shift selection

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
## Related Issue(s)

closes https://github.com/SAP/fundamental-ngx/issues/7622
closes https://github.com/SAP/fundamental-ngx/issues/7650

## Description
- An ability to select ranges is not implemented in multi-select. 
- An ability to select ranges is not reflected in core-table examples. Also "indeterminate" state is not applied, when some rows are selected.

Part of #7618

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/table/examples/table-checkboxes-example.component.html [hunk 1]
```diff
@@ -3,7 +3,11 @@
     <thead fd-table-header>
         <tr fd-table-row>
             <th fd-table-cell>
-                <fd-checkbox [ngModel]="checkboxValue" (ngModelChange)="selectMaster($event, 'cozy')"></fd-checkbox>
+                <fd-checkbox
+                    [tristate]="true"
+                    [ngModel]="checkboxValue"
+                    (ngModelChange)="selectMaster($event, 'cozy')"
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/table/examples/table-checkboxes-example.component.html:9]: Please change `master` to `main`. Thank you

File: libs/core/src/lib/checkbox/checkbox/checkbox.component.ts [hunk 2]
```diff
@@ -57,6 +56,14 @@ export class CheckboxComponent implements ControlValueAccessor, OnInit, OnDestro
     @Input()
     ariaLabel = '';
 
+    @Input()
```
Comment (InnaAtanasova) [inline libs/core/src/lib/checkbox/checkbox/checkbox.component.ts:59]: Please add typedocs

**Review by InnaAtanasova — COMMENTED** [review]
Just 2 minor comments, but o.w. great job

**Review by N1XUS — APPROVED** [review]

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Nice work with the shift feature, but this PR has broken a piece of platform's multi-combobox, I cannot click the check mark for the items in the dropdown to select them

**Comment by dmitry-stepanenko** [comment]
@mikerodonnell89 thank you, fixed

**Review by platon-rov — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Review by aberikashvili — APPROVED** [review]
LGTM

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
@dmitry-stepanenko I apologize for being so nit-picky but the functionality has changed for platform multi-combobox, previously when clicking the checkbox, the popover would stay open, it would only close if you clicked the label.  Now clicking the checkbox also closes the popover

**Pull request merged**